gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTableMetadataValidatorUtilsTest.java

```diff
@@ -0,0 +1,200 @@
+    );
+  }
+
+  @Test
+  public void testValidatePartitionSpecWithDiffName() {
+    PartitionSpec partitionSpec12 = PartitionSpec.builderFor(schema1)
+        .identity("field2")
+        .build();
+    TableMetadata tableMetadataWithSchema1AndPartitionSpec12 = TableMetadata.newTableMetadata(schema1, partitionSpec12,
+        "tableLocationForSchema1WithPartitionSpec12", new HashMap<>());
+    verifyFailUnlessCompatibleStructureIOException(tableMetadataWithSchema1AndPartitionSpec1,
+        tableMetadataWithSchema1AndPartitionSpec12, PARTITION_SPEC_MISMATCH_EXCEPTION);
+  }
+
+  @Test
+  public void testValidatePartitionSpecWithUnpartitioned() {
+    verifyFailUnlessCompatibleStructureIOException(tableMetadataWithSchema1AndUnpartitionedSpec,
+        tableMetadataWithSchema1AndPartitionSpec1, PARTITION_SPEC_MISMATCH_EXCEPTION);
+  }
+
+  @Test
+  public void testPartitionSpecWithDifferentTransform() {
+    PartitionSpec partitionSpec12 = PartitionSpec.builderFor(schema1)
+        .truncate("field1", 4)
+        .build();
+    TableMetadata tableMetadataWithSchema1AndPartitionSpec12 = TableMetadata.newTableMetadata(schema1, partitionSpec12,
+        "tableLocationForSchema1WithPartitionSpec12", new HashMap<>());
+    verifyFailUnlessCompatibleStructureIOException(tableMetadataWithSchema1AndPartitionSpec1,
+        tableMetadataWithSchema1AndPartitionSpec12, PARTITION_SPEC_MISMATCH_EXCEPTION);
+  }
+
+  @Test
+  public void testStrictPartitionSpecEquality() throws IOException {
+    PartitionSpec partitionSpecWithTwoCols = PartitionSpec.builderFor(schema1)
+        .identity("field1")
+        .identity("field2")
+        .build();
+
+    TableMetadata tableMetadataWithSchema1AndPartitionSpecWithTwoCols = TableMetadata.newTableMetadata(schema1,
+        partitionSpecWithTwoCols, "tableLocationForSchema1WithPartitionSpecWithTwoCols", new HashMap<>());
+    TableMetadata updatedMetadataForTableMetadataWithSchema1AndPartitionSpec1 = tableMetadataWithSchema1AndPartitionSpec1
+        .updatePartitionSpec(tableMetadataWithSchema1AndPartitionSpecWithTwoCols.spec());
+
+    IcebergTableMetadataValidatorUtils.failUnlessCompatibleStructure(
+        tableMetadataWithSchema1AndPartitionSpecWithTwoCols,
+        updatedMetadataForTableMetadataWithSchema1AndPartitionSpec1,
+        VALIDATE_STRICT_PARTITION_EQUALITY_FALSE);
+
+    verifyFailUnlessCompatibleStructureIOException(tableMetadataWithSchema1AndPartitionSpecWithTwoCols,
+        updatedMetadataForTableMetadataWithSchema1AndPartitionSpec1, PARTITION_SPEC_MISMATCH_EXCEPTION);
+  }
+
+  private void verifyFailUnlessCompatibleStructureIOException(TableMetadata tableAMetadata,
+      TableMetadata tableBMetadata, String expectedMessage) {
+    IOException exception = Assert.expectThrows(IOException.class, () -> {
+      IcebergTableMetadataValidatorUtils.failUnlessCompatibleStructure(tableAMetadata, tableBMetadata,
+          VALIDATE_STRICT_PARTITION_EQUALITY_TRUE);
+    });
```

Comment:
because strict equality is hard-coded, shall we rename to sth like: `verifyStrictFailUnlessCompabitleStructureThrows`